src/fmureceiver.cpp

```diff
@@ -0,0 +1,307 @@
+#include "fmureceiver.h"
+
+
+
+/**
+*  local functions
+*/
+static void fmuWrapperLogger(jm_callbacks* c, jm_string module, jm_log_level_enu_t log_level, jm_string message)
+{
+    switch (log_level)
+    {
+        case jm_log_level_fatal:
+            qDebug() << (message);
+            break;
+        case jm_log_level_error:
+        case jm_log_level_warning:
+            qDebug() << message;
+            break;
+        case jm_log_level_nothing:
+        case jm_log_level_info:
+        case jm_log_level_verbose:
+        case jm_log_level_debug:
+        case jm_log_level_all:
+            qDebug() << message;
+            break;
+    }
+}
+
+static void* decode_integer_to_pointer(fmi2_integer_t hi, fmi2_integer_t lo)
+{
+#if PTRDIFF_MAX == INT64_MAX
+    union addrconv
+    {
+        struct
+        {
+            int lo;
+            int hi;
+        } base;
+        unsigned long long address;
+    } myaddr;
+    myaddr.base.lo = lo;
+    myaddr.base.hi = hi;
+    return reinterpret_cast<void*>(myaddr.address);
+#elif PTRDIFF_MAX == INT32_MAX
+    return reinterpret_cast<void*>(lo);
+#else
+#error "Cannot determine 32bit or 64bit environment!"
+#endif
+}
+
+FMUReceiver::FMUReceiver()
+    : IMessageSource()
+
+    , isRunning_(false)
+    , isThreadTerminated_(false)
+    , currentDataType_(DataType::Groundtruth)
+{
+}
+
+void FMUReceiver::ConnectRequested(const QString &ipAddress, const QString &port, const QString& fmuPath, DataType dataType)
+{
+    if (!fmuPath.isEmpty())
+    {
+        logLevel_ = LogLevel::Warn;
+        ip_ = ipAddress.toStdString();
+        port_ = port.toStdString();
+        FMUPath_ = fmuPath.toStdString();
+        tmpPath_ = FMUPath_.substr(0, FMUPath_.find_last_of("/\\"));
+
+        // Start initialization
+        if (!initializeFMUWrapper())
+        {
+            QString message = "FMU Wrapper initialization failed!";
+            emit Disconnected(message);
+            return;
+        }
+
+        fmu_ = fmi2_import_parse_xml(context_, tmpPath_.c_str(), 0);
+
+        if (!fmu_)
+        {
+            QString message = "Error parsing modeldescription.xml of fmu ";
+            emit Disconnected(message);
+            return;
+        }
+
+        if (fmi2_import_get_fmu_kind(fmu_) == fmi2_fmu_kind_me)
```

Comment:
Test should follow logic in warning message, i.e. test for != fmi2_fmu_kind_cs, rather than me.